general/forall.hpp

```diff
@@ -87,58 +87,94 @@ void OmpWrap(const int N, HBODY &&h_body)
-void RajaCudaWrap1D(const int N, DBODY &&d_body)
+void RajaDeviceWrap1D(const int N, DBODY &&d_body)
 {
    //true denotes asynchronous kernel
-   RAJA::forall<RAJA::cuda_exec<BLOCKS,true>>(RAJA::RangeSegment(0,N),d_body);
+#if defined(RAJA_CUDA_ACTIVE)
+   using ForPolicy = RAJA::cuda_exec<MFEM_CUDA_BLOCKS,true>;
+#else
+   using ForPolicy = RAJA::hip_exec<MFEM_CUDA_BLOCKS,true>;
+#endif
+   RAJA::forall<ForPolicy>(RAJA::RangeSegment(0,N),d_body);
 }
 
 template <typename DBODY>
-void RajaCudaWrap2D(const int N, DBODY &&d_body,
-                    const int X, const int Y, const int BZ)
+void RajaDeviceWrap2D(const int N, DBODY &&d_body,
+                      const int X, const int Y, const int BZ)
 {
    MFEM_VERIFY(N>0, "");
    MFEM_VERIFY(BZ>0, "");
    const int G = (N+BZ-1)/BZ;
-   RAJA::kernel<RAJA::KernelPolicy<
-   RAJA::statement::CudaKernelAsync<
-   RAJA::statement::For<0, RAJA::cuda_block_x_direct,
-        RAJA::statement::For<1, RAJA::cuda_thread_x_direct,
-        RAJA::statement::For<2, RAJA::cuda_thread_y_direct,
-        RAJA::statement::For<3, RAJA::cuda_thread_z_direct,
-        RAJA::statement::Lambda<0, Segs<0>>>>>>>>>
-        (RAJA::make_tuple(RAJA::RangeSegment(0,G), RAJA::RangeSegment(0,X),
-                          RAJA::RangeSegment(0,Y), RAJA::RangeSegment(0,BZ)),
-         [=] RAJA_DEVICE (const int n)
+
+   RAJA::expt::launch<launch_policy>
+   (RAJA::expt::DEVICE,
+    RAJA::expt::Resources(RAJA::expt::Teams(G), RAJA::expt::Threads(X, Y, BZ)),
+    [=] RAJA_DEVICE (RAJA::expt::LaunchContext ctx)
    {
-      const int k = n*BZ + threadIdx.z;
-      if (k >= N) { return; }
-      d_body(k);
+
+      RAJA::expt::loop<teams_x>(ctx, RAJA::RangeSegment(0, G), [&] (const int n)
+      {
+
+         RAJA::expt::loop<threads_z>(ctx, RAJA::RangeSegment(0, BZ), [&] (const int tz)
+         {
+
+            const int k = n*BZ + tz;
+            if (k >= N) { return; }
+            d_body(k);
+
+         });
+
+      });
+
    });
+
+#if defined(RAJA_CUDA_ACTIVE)
    MFEM_GPU_CHECK(cudaGetLastError());
+#else
+   MFEM_GPU_CHECK(hipGetLastError());
+#endif
 }
 
 template <typename DBODY>
-void RajaCudaWrap3D(const int N, DBODY &&d_body,
-                    const int X, const int Y, const int Z)
+void RajaDeviceWrap3D(const int N, DBODY &&d_body,
+                      const int X, const int Y, const int Z)
 {
    MFEM_VERIFY(N>0, "");
-   RAJA::kernel<RAJA::KernelPolicy<
-   RAJA::statement::CudaKernelAsync<
-   RAJA::statement::For<0, RAJA::cuda_block_x_direct,
-        RAJA::statement::For<1, RAJA::cuda_thread_x_direct,
-        RAJA::statement::For<2, RAJA::cuda_thread_y_direct,
-        RAJA::statement::For<3, RAJA::cuda_thread_z_direct,
-        RAJA::statement::Lambda<0, Segs<0>>>>>>>>>
-        (RAJA::make_tuple(RAJA::RangeSegment(0,N), RAJA::RangeSegment(0,X),
-                          RAJA::RangeSegment(0,Y), RAJA::RangeSegment(0,Z)),
-   [=] RAJA_DEVICE (const int k) { d_body(k); });
+
```

Comment:
Add `using Raja::expt;`?

Reply:
Done! Only the items related to RAJA Teams is under the ::expt name space.